<?xml version="1.0"?>
<launch>
  <arg name="namespace" default="flir_boson"/>
  <arg name="frame_id" default="boson_camera"/>

  <arg name="left_ip" default="169.254.87.157"/>
  <arg name="right_ip" default="169.254.82.142"/>
    
  <!-- valid values are 30.0 or 60.0 for Bosons -->
  <arg name="frame_rate" default="60.0"/>

  <!-- location of the camera calibration file -->
  <arg name="camera_info_url" default="package://flir_boson_ethernet/example_calibrations/BlackFlyS.yaml"/>

  <node pkg="flir_boson_ethernet" 
        type="flir_boson_ethernet_sync_node" 
        name="flir_boson_ethernet_sync_node" 
        ns="$(arg namespace)"
        output="screen">
    <param name="frame_id" type="str" value="$(arg frame_id)"/>
    <param name="left_ip" type="str" value="$(arg left_ip)"/>
    <param name="right_ip" type="str" value="$(arg right_ip)"/>
    <param name="frame_rate" type="double" value="$(arg frame_rate)"/>
    <param name="camera_info_url" type="str" value="$(arg camera_info_url)"/>
  </node>

  <node name="image_view" pkg="image_view" type="image_view" respawn="false" output="screen">
    <remap from="image" to="flir_boson/image_raw" />
  </node>
</launch>